priv/tools/internal/offline_delete.erl

```diff
@@ -0,0 +1,143 @@
+%% turn off AAE on other nodes that's running on the cluster.
+
+main(["--dry-run", BitcaskDir, BlocksListFile]) ->
+    offline_delete(BitcaskDir, BlocksListFile, true);
+main([BitcaskDir, BlocksListFile]) ->
+    offline_delete(BitcaskDir, BlocksListFile, false);
+main(_) ->
+    io:format(standard_error,
+              "options: [--dry-run] <BitcaskDir> <BlocksListFile>~n", []).
+
+-spec open_all_bitcask(filename:filename()) ->
+                              orddict:orddict(non_neg_integer(), reference()).
+open_all_bitcask(BitcaskDir) ->
+    {ok, List} = file:list_dir(BitcaskDir),
+    Result = lists:map(fun(File) ->
+                               Filename = filename:join(BitcaskDir, File),
+                               case bitcask:open(Filename, [read_write]) of
+                                   Ref when is_reference(Ref) ->
+                                       {list_to_integer(File), Ref};
+                                   Other ->
+                                       error({File, Other})
+                               end
+                       end, List),
+    orddict:from_list(Result).
+
+-spec close_all_bitcask(orddict:orddict(non_neg_integer(), reference())) -> ok.
+close_all_bitcask(Bitcasks) ->
+    orddict:map(fun(_, Ref) ->
+                        bitcask:close(Ref)
+                end, Bitcasks).
+
+offline_delete(BitcaskDir, BlocksListFile, DryRun) ->
+    {ok, Fd} = file:open(BlocksListFile, [read]),
+    BC = open_all_bitcask(BitcaskDir),
+    io:format(standard_error, "~p bitcask directories at ~s opened.~n",
+              [length(BC), BitcaskDir]),
+    {ok, Deleted} = for_each_line(Fd, BC, DryRun, 0),
+    %% io:format(standard_error, "~p~n", [BC]),
+    io:format(standard_error, "~p blocks at ~s was deleted"
+              " (dry run: ~p).~n",
+              [Deleted, BitcaskDir, DryRun]),
+    close_all_bitcask(BC),
+    ok = file:close(Fd).
+
+for_each_line(Fd, BC, DryRun, Count) ->
+    case Count rem 1000 of
+        500 ->
+            io:format(standard_error,
+                      "~p blocks has been deleted.~n",
+                      [Count]);
+        _ ->
+            noop
+    end,
+    case file:read_line(Fd) of
+        {ok, Line} ->
+            Tokens = string:tokens(Line, "\t \n"),
+            [V1, V2, V3, B, K, _UUIDStr, _SeqNo] = Tokens,
+            Bucket = mochihex:to_bin(B),
+            Key = mochihex:to_bin(K),
+            %% io:format("trying ~p~n", [{list_to_integer(V1),
+            %%                            list_to_integer(V2),
+            %%                            list_to_integer(V3),
+            %%                            UUIDStr,
+            %%                            list_to_integer(SeqNo)}]),
+            delete(BC, list_to_integer(V1), Bucket, Key, DryRun),
+            delete(BC, list_to_integer(V2), Bucket, Key, DryRun),
+            delete(BC, list_to_integer(V3), Bucket, Key, DryRun),
+            for_each_line(Fd, BC, DryRun, Count+3);
```

Comment:
`delete/5` may skip bitcask delete operation. If actual `bitcask:delete` ops are counted, it can be used for sanity check. (almost for fool-proof, for example, wrong bitcask directory is specified in command line and all delete ops are skipped)